tools/cold-backup/run.sh

```diff
@@ -0,0 +1,481 @@
+        fi
+        echo "Creating pvc $pvc of size $size"
+        sed "$exp" $__HOME/templates/pvc.yaml | kubectl -n $NAMESPACE create -f -
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+__compress() {
+    local source_dir="$1"
+    local target_file="$2"
+    if [[ -n $source_dir && -n $target_file ]]; then
+        local current_dir=$(pwd)
+        local target_path=$(readlink -f $target_file)
+        local target_name=$(basename "$target_file")
+        cd $source_dir
+        echo "Compressing $source_dir to $target_file"
+        zip -FSqr $target_name *
+        mv $target_name $target_path
+        cd $current_dir
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+__uncompress() {
+    local source_file="$1"
+    local readonly target_dir="$2"
+    if [[ -n $source_file && -n $target_dir ]]; then
+        echo "Uncompressing $source_file to $target_dir"
+        rm -rf $target_dir
+        mkdir -p $target_dir
+        unzip -qo $source_file -d $target_dir
+        chmod -R ugo+rwx $target_dir
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+backup() {
+    # init context
+    local readonly tmp="$__TMP/$NAMESPACE/$CLUSTER_NAME"
+    if [[ $CONFIRM == true ]]; then
+        echo "Backup of cluster $NAMESPACE/$CLUSTER_NAME"
+        echo "The cluster won't be available for the entire duration of the process"
+        __confirm
+    fi
+    if [[ $INCREMENTAL == true ]]; then
+        echo "Starting incremental backup"
+    else
+        echo "Starting full backup"
+        rm -rf $tmp
+    fi
+    mkdir -p $tmp/resources $tmp/data
+    __export_env $CLUSTER_NAME $tmp/env
+    __stop_cluster $CLUSTER_NAME
+
+    # export resources
+    __export_res kafka $CLUSTER_NAME $tmp/resources
+    __export_res kafkatopics strimzi.io/cluster=$CLUSTER_NAME $tmp/resources
+    __export_res kafkausers strimzi.io/cluster=$CLUSTER_NAME $tmp/resources
+    __export_res kafkarebalances strimzi.io/cluster=$CLUSTER_NAME $tmp/resources
+    # cluster configmap and secrets
+    __export_res configmaps app.kubernetes.io/instance=$CLUSTER_NAME $tmp/resources
+    __export_res secrets app.kubernetes.io/instance=$CLUSTER_NAME $tmp/resources
+    # custom configmap and secrets
+    if [[ -n $CUSTOM_CM ]]; then
+        for name in $(printf $CUSTOM_CM | sed "s/,/ /g"); do
+            __export_res configmap $name $tmp/resources
+        done
+    fi
+    if [[ -n $CUSTOM_SE ]]; then
+        for name in $(printf $CUSTOM_SE | sed "s/,/ /g"); do
+            __export_res secret $name $tmp/resources
+        done
+    fi
+
+    # for each PVC, rsync data from PV to backup
+    for ((i = 0; i < $ZK_REPLICAS; i++)); do
+        local pvc="data-$CLUSTER_NAME-zookeeper-$i"
+        local path="$tmp/data/$pvc"
+        mkdir -p $path
+        __rsync $pvc $path
+    done
+    for ((i = 0; i < $KAFKA_REPLICAS; i++)); do
+        if ((JBOD_VOL_NUM > 1)); then
```

Comment:
What if I have JBOD storage with only one volume? This is probably why it does not work for me.

Reply:
OK